packages/services/src/fees/testing.rs

```diff
@@ -0,0 +1,90 @@
+use std::{collections::BTreeMap, ops::RangeInclusive};
+
+use itertools::Itertools;
+
+use super::{Api, Fees, FeesAtHeight, SequentialBlockFees};
+
+#[derive(Debug, Clone, Copy)]
+pub struct ConstantFeeApi {
+    fees: Fees,
+}
+
+impl ConstantFeeApi {
+    pub const fn new(fees: Fees) -> Self {
+        Self { fees }
+    }
+}
+
+impl Api for ConstantFeeApi {
+    async fn fees(&self, height_range: RangeInclusive<u64>) -> crate::Result<SequentialBlockFees> {
+        let fees = height_range
+            .into_iter()
+            .map(|height| FeesAtHeight {
+                height,
+                fees: self.fees,
+            })
+            .collect_vec();
+
+        Ok(fees.try_into().unwrap())
+    }
+
+    async fn current_height(&self) -> crate::Result<u64> {
+        Ok(0)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct PreconfiguredFeeApi {
+    fees: BTreeMap<u64, Fees>,
+}
+
+impl Api for PreconfiguredFeeApi {
+    async fn current_height(&self) -> crate::Result<u64> {
+        Ok(*self
+            .fees
+            .keys()
+            .last()
+            .expect("no fees registered with PreconfiguredFeesProvider"))
+    }
+
+    async fn fees(&self, height_range: RangeInclusive<u64>) -> crate::Result<SequentialBlockFees> {
+        let fees = self
+            .fees
+            .iter()
+            .skip_while(|(height, _)| !height_range.contains(height))
+            .take_while(|(height, _)| height_range.contains(height))
+            .map(|(height, fees)| FeesAtHeight {
+                height: *height,
+                fees: *fees,
+            })
+            .collect_vec();
+
+        Ok(fees.try_into().expect("block fees not sequential"))
+    }
+}
+
+impl PreconfiguredFeeApi {
+    pub fn new(blocks: impl IntoIterator<Item = (u64, Fees)>) -> Self {
+        Self {
+            fees: blocks.into_iter().collect(),
+        }
+    }
+}
+
+pub fn incrementing_fees(num_blocks: u64) -> SequentialBlockFees {
```

Comment:
this one doesn't seem to be used anywhere

Reply:
[2daea6c](https://github.com/FuelLabs/fuel-block-committer/commit/2daea6c7bf0e67737a90383fc3a74e3d53966fcd)